src/tooltip/test/tooltip.spec.js

```diff
@@ -277,6 +278,52 @@ describe('tooltip', function() {
 
   });
 
+  describe('with specified popup close delay', function () {
+
+    beforeEach(inject(function ($compile) {
+      scope.delay='1000';
+      elm = $compile(angular.element(
+        '<span tooltip="tooltip text" tooltip-popup-delay="1000"' +
+        'tooltip-popup-close-delay="{{delay}}">Selector Text</span>'
+      ))(scope);
+      elmScope = elm.scope();
+      tooltipScope = elmScope.$$childTail;
+      scope.$digest();
+    }));
+
+    it('should open after timeout and close after the close delay timeout', inject(function ($timeout) {
+
+      elm.trigger('mouseenter');
```

Comment:
You are welcome :)

It can be impressive to disable [the presently included `jQuery`](https://github.com/angular-ui/bootstrap/blob/master/karma.conf.js#L18) and see 425 tests out of 755 failing... no small task to suggest!

As an attempt to make it more practical and less daunting, my suggestion would be to state it in the contribution guidelines as *recommendation* for new PRs to consider. It can be always documented in then comments and linked to this or other discussions if that gets too long. 

So sure, are welcome to open issue referring to this and see what the consensus is.